examples/regression/1-quickstart/plot_regression_housing_dataset.py

```diff
@@ -0,0 +1,426 @@
+
+
+df = pd.concat([X, y], axis=1)
+pear_corr = df.corr(method='pearson')
+pear_corr.style.background_gradient(cmap='Greens', axis=0)
+
+
+##############################################################################
+# Now let's visualize a histogram of the price of the houses.
+
+
+fig, axs = plt.subplots(1, 1, figsize=(5, 5))
+axs.hist(y, bins=50)
+axs.set_xlabel("Median price of houses")
+axs.set_title("Histogram of house prices")
+axs.xaxis.set_major_formatter(FormatStrFormatter('%.0f'+"k"))
+plt.show()
+
+
+##############################################################################
+# Let's now create the different splits for the dataset, with a training,
+# calibration and test set.
+
+
+X_train, X_test, y_train, y_test = train_test_split(
+    X,
+    y['MedHouseVal'],
+    random_state=random_state
+)
+X_train, X_calib, y_train, y_calib = train_test_split(
+    X_train,
+    y_train,
+    random_state=random_state
+)
+
+
+##############################################################################
+# 2. Optimizing estimator
+# --------------------------------------------------------------------------
+# Optimization of the `LGBMRegressor` using `RandomizedSearchCV` to find the
+# optimal model to predict the house prices.
+
+
+estimator = LGBMRegressor(
+    objective='quantile',
+    alpha=0.5,
+    random_state=random_state
+)
+params_distributions = dict(
+    num_leaves=randint(low=10, high=50),
+    max_depth=randint(low=3, high=20),
+    n_estimators=randint(low=50, high=300),
+    learning_rate=uniform()
+)
+optim_model = RandomizedSearchCV(
+    estimator,
+    param_distributions=params_distributions,
+    n_jobs=-1,
+    n_iter=100,
+    cv=KFold(n_splits=5),
+    verbose=-1
+)
+optim_model.fit(X_train, y_train)
+estimator = estimator.set_params(**optim_model.best_params_)
+
+
+##############################################################################
+# 3. Comparaison of MAPIE methods
+# --------------------------------------------------------------------------
+# We will now proceed to compare the different regression method available
+# on MAPIE. For this tutorial we will compare the "naive", "jackknife plus ab",
+# "cv plus" and "conformalized quantile regression".
+
+
+def sort(
+    y_test: NDArray, y_pred: NDArray, y_pis: NDArray
+) -> Tuple[NDArray, NDArray, NDArray, NDArray]:
+    """
+    Sorting the dataset such that you can make better plots.
+
+    Parameters
+    ----------
+    y_test : NDArray
+        _description_
+    y_pred : NDArray
+        _description_
+    y_pis : NDArray
+        _description_
+
+    Returns
+    -------
+    Tuple[NDArray, NDArray, NDArray, NDArray]
+        _description_
+    """
+    indices = np.argsort(y_test)
+    y_test_sorted = np.array(y_test)[indices]
+    y_pred_sorted = y_pred[indices]
+    y_lower_bound = y_pis[:, 0, 0][indices]
+    y_upper_bound = y_pis[:, 1, 0][indices]
+    return y_test_sorted, y_pred_sorted, y_lower_bound, y_upper_bound
+
+
+def plot(
```

Comment:
Give the function a more explicit name like "plot_prediction_intervals".

Reply:
done